core/src/retransmit_stage.rs

```diff
@@ -224,57 +175,51 @@ fn maybe_reset_shreds_received_cache(
 
 #[allow(clippy::too_many_arguments)]
 fn retransmit(
+    thread_pool: &ThreadPool,
     bank_forks: &RwLock<BankForks>,
     leader_schedule_cache: &LeaderScheduleCache,
     cluster_info: &ClusterInfo,
-    shreds_receiver: &Mutex<mpsc::Receiver<Vec<Shred>>>,
-    sock: &UdpSocket,
-    id: u32,
-    stats: &RetransmitStats,
+    shreds_receiver: &mpsc::Receiver<Vec<Shred>>,
+    sockets: &[UdpSocket],
+    stats: &mut RetransmitStats,
     cluster_nodes_cache: &ClusterNodesCache<RetransmitStage>,
-    hasher_reset_ts: &AtomicU64,
+    hasher_reset_ts: &mut Instant,
     shreds_received: &Mutex<ShredFilterAndHasher>,
     max_slots: &MaxSlots,
     first_shreds_received: &Mutex<BTreeSet<Slot>>,
     rpc_subscriptions: Option<&RpcSubscriptions>,
-) -> Result<()> {
+) -> Result<(), RecvTimeoutError> {
     const RECV_TIMEOUT: Duration = Duration::from_secs(1);
-    let shreds_receiver = shreds_receiver.lock().unwrap();
     let mut shreds = shreds_receiver.recv_timeout(RECV_TIMEOUT)?;
     let mut timer_start = Measure::start("retransmit");
-    while let Ok(more_shreds) = shreds_receiver.try_recv() {
-        shreds.extend(more_shreds);
-        if shreds.len() >= MAX_SHREDS_BATCH_SIZE {
-            break;
-        }
-    }
-    drop(shreds_receiver);
+    shreds.extend(shreds_receiver.try_iter().flatten());
```

Comment:
Is there no longer a need to limit the batch size received before starting to process the shreds?

Reply:
With the previous code, one thread was locking the channel and, without a batch-size limit, if there was a burst of shreds on the channel, they would all be assigned to that single thread. So the batch-size limit was mitigating unbalanced workload distribution across the threads. Here, we don't have that issue anymore because assigning shreds to threads is done after consuming from the channel.

The other aspect is that if adding a batch-size limit will improve latencies for shreds processing; 
I think it will actually do more harm than good. Because rayon works better with larger batches (due to fixed-cost amortization), and adding a batch-size limit will force rayon to work sub-optimally causing overall larger average latencies when there is a heavy load by cutting the batches short.